### PR TITLE
[#786] OL Loading indicator should only be affected by last OL request

### LINKED ISSUE #786: OL Loading indicator should only be affected by last OL request
## How to reproduce

1. Select product and scene
2. Change scenes fast using next / previous button
3. OL loading indicator disappears, but the scene is loading

## Proposed solution

OL handlers for starting / ending request should be context aware, and context should be reset when new request is made.
(loading finish handler should set loading to false only for the last request)

### PR DESCRIPTION
**Pull request by danielkryska**
- Change Image Wms to Tile Wms loading handling

Closes #786

### CODE REVIEW

File: s4e-web/src/app/views/map-view/state/overlay/overlay.model.ts [hunk 1]
```diff
@@ -61,33 +61,14 @@ export function convertToUIOverlay(
...
-  InjectorModule
-    .Injector
-    .get(NotificationService)
-    .addGeneral({
-      type: 'error',
-      content: 'Wczytanie sceny nie powiodło się'
-    });
+  const tilesLoaderLoader = new TilesLoaderLoader(source);
+  tilesLoaderLoader.start$.then(() => loaderService.startBackground());
+  tilesLoaderLoader.end$.then(() => loaderService.stopBackground());
```
Comment (jswk) [inline s4e-web/src/app/views/map-view/state/overlay/overlay.model.ts:69]: From my view, once this typo (double loader) is fixed then the PR is good to go

**Review by jswk — CHANGES_REQUESTED** [review]

**Comment by michal-szostak** [comment]
I have a question - does this PR solves the issue which it is connected to?
Because I indeed see change from `Image` to `Tile`, but I do not see anything which would solve the #786.
(The PR instance is down, so I couldn't check there)

**Comment by danielkryska** [comment]
@michal-szostak I belive this was the issue. 

Before we were using `OL` image hooks like `imageloadend` and `imageloadstart` for `ImageWms` usage only.
```
    source.once('imageloadend', () => resolve());
    source.once('imageloadstart', () => resolve());
```

, but then @jswk changed handling to `Tiles`, so finally loading methods names should be changed also. In this case to `tileloadend` and `tileloadstart`. Loading indicator we've seen `sometimes` while choosing `product` was misleading and the bug wasn't connected with unsloved stream/promise/async.
 
```
    this.end$ = new Promise((resolve) => source.once('tileloadend', () => resolve()));
    this.start$ = new Promise((resolve) => source.once('tileloadstart', () => resolve()));

**Comment by danielkryska** [comment]
@michal-szostak I've updated handling of tiles loader to switch map to new incoming promise. It's working much smoother than before.

**Review by jswk — APPROVED** [review]
Now it works correctly. Thanks for fixing this :)

**Review by michal-szostak — APPROVED** [review]

**Pull request merged**